### PR TITLE
DOC SQL placeholders

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-framework/issues/10860

### CODE REVIEW

File: en/04_Changelogs/5.1.0.md
Comment (ChloeSartorelli) [inline en/04_Changelogs/5.1.0.md:1]: This is really confusing... it makes it sound like I can't do this:

```php
$placeholders = DB::placeholders($ids);
$list->where('MyIDField IN (' . $placeholders . ')');
```

I think this should be a lot more succinct, and not sound like we're removing some functionality. Something along the lines of:
````md
### Optimised queries when filtering against IDs {#filter-by-ids}

`DataList` queries filtering against a list of IDs have been optimised when all of the following criteria are met:
- the column being filtered is a [`DBPrimarykey`](api:SilverStripe\ORM\FieldType\DBPrimaryKey) or a [`DBForiegnKey`](api:SilverStripe\ORM\FieldType\DBForiegnKey)
- the values being filtered are all either integers or valid integer strings
- using placeholders for integer ids has been configured off, which is the default config value.

If you want to disable this optimisation you can do so with this configuration:

```yml
SilverStripe\ORM\Filters\ExactMatchFilter:
  use_placeholders_for_integer_ids: true
```
````

File: en/02_Developer_Guides/09_Security/08_SQL_Placeholders.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+title: SQL Placeholders
+summary: SQL placeholders in ORM queries
+icon: tachometer-alt
+---
+
+# SQL Placeholders
+
+SQL placeholders are `?` characters used as a placeholder for a value in a SQL query as a way to prevent SQL injection attacks. They are used by default extensively in queries created by the ORM.
+
+Placeholders have been disabled for performance reasons when filtering by an array of integer only values on a column that is either a [DBPrimarykey](api:SilverStripe\ORM\FieldType\DBPrimaryKey) or a [DBForiegnKey](api:SilverStripe\ORM\FieldType\DBForiegnKey). An example of this type of ORM filter is `->filter(['ID' => $ids])` which will turn into a SQL containing `WHERE IN (<ids>)`.
```
Comment (ChloeSartorelli) [inline en/02_Developer_Guides/09_Security/08_SQL_Placeholders.md]: Please don't use "placeholders have been disabled" - nothing has been disabled. We're simply not using them for those queries.

Comment (ChloeSartorelli) [inline en/02_Developer_Guides/09_Security/08_SQL_Placeholders.md]:
```suggestion
Placeholders have been disabled for performance reasons when filtering by an array of integer only values on a column that is either a [`DBPrimarykey`](api:SilverStripe\ORM\FieldType\DBPrimaryKey) or a [`DBForiegnKey`](api:SilverStripe\ORM\FieldType\DBForiegnKey). An example of this type of ORM filter is `->filter(['ID' => $ids])` which will turn into a SQL containing `WHERE IN (<ids>)`. 
```

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: en/02_Developer_Guides/09_Security/08_SQL_Placeholders.md (hunk 1 above)
Comment (emteknetnz) [reply en/02_Developer_Guides/09_Security/08_SQL_Placeholders.md]: Updated

Comment (emteknetnz) [reply en/02_Developer_Guides/09_Security/08_SQL_Placeholders.md]: Updated

File: en/04_Changelogs/5.1.0.md
Comment (emteknetnz) [reply en/04_Changelogs/5.1.0.md:1]: Updated

File: en/04_Changelogs/5.1.0.md [hunk 2]
```diff
@@ -11,6 +11,7 @@ title: 5.1.0 (unreleased)
   - [GraphQL schema can be stored in `silverstripe-cache`](#gql-schema-is-cache)
   - [Improvement to page search performance with Elemental](#cms-search-performance)
   - [New `InheritedPermissions` option - only these members](#only-these-members)
+  - [SQL placeholders disabled for WHERE IN ID queries](#placeholders-disabled)
```
Comment (ChloeSartorelli) [inline en/04_Changelogs/5.1.0.md]:
```suggestion
  - [Optimised queries when filtering against IDs](#filter-by-ids)
```

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: en/04_Changelogs/5.1.0.md (hunk 2 above)
Comment (emteknetnz) [reply en/04_Changelogs/5.1.0.md]: Updated

**Pull request merged**